src/skmatter/_selection.py

```diff
@@ -63,14 +63,16 @@ class GreedySelector(SelectorMixin, MetaEstimatorMixin, BaseEstimator):
     progress_bar: bool, default=False
               option to use `tqdm <https://tqdm.github.io/>`_
               progress bar to monitor selections. Stored in
-              :py:attr:`self.report_progress`.
+              :py:attr:`self.report_progress_`.
 
     full : bool, default=False
         In the case that all non-redundant selections are exhausted, choose
         randomly from the remaining features. Stored in :py:attr:`self.full`.
 
     random_state: int or RandomState instance, default=0
 
+    axis: [0,1] axis over which we are doing selection
```

Comment:
Isn't `axis` and `selection_type` redundant? They seem to me doing more or less the same thing.

Shouldn't be `axis` be marked as private? Since it determines the control flow in the fit and transform function. A public member variable indicates that you can change it whenever you want. But changing the axis after fitting, is changing the internal logic in transform and therefore most likely causes errors. Have you thought about this?
(Also same goes for src/skmatter/sample_selection/_voronoi_fps.py)

Reply:
Sklearn prefers that parameters initialized in the constructor be public, and private ones be set during fitting. Including the underscore causes an error in their estimator check, unfortunately

Reply:
I kept both for backwards compatibility, otherwise we'd need to increment the version.

Reply:
> Sklearn prefers that parameters initialized in the constructor be public, and private ones be set during fitting. Including the underscore causes an error in their estimator check, unfortunately

But before the PR `axis` was private and first time defined in the fit function. That is in agreement of what you write.

> I kept both for backwards compatibility, otherwise we'd need to increment the version.

I am not sure why we need to increment the version, because axis was not a public before. Making it public, makes it available to the user and makes it more annoying to remove it later.

Reply:
All done! I made sure everything flies with the changes in #196 as well.